enums/message_ state.go

```diff
@@ -0,0 +1,22 @@
+package enums
+
+type MessageState uint8
+
+const (
+	MessageStateInvalid MessageState = iota
+	MessageReceived
+	MessageProcessing
+	MessageProcessed
+)
+
+func (s MessageState) String() string {
+	switch s {
+	case MessageReceived:
+		return "MessageReceived"
+	case MessageProcessing:
+		return "MessageProcessing"
+	case MessageProcessed:
```

Comment:
Whether the message is processed successfully or fails, the state is the state  both processed?

Reply:
a failed state is needed, i will add it in next pr